V1-1

JeGoi opened the pull request:
Add debug mode
Clean functions
Clean exit
Clean few other stuff
Remove fatal
Move print stdout to debug
Etc.

JeGoi committed:
Add debug

JeGoi committed:
Add debug in certificate

JeGoi committed:
Add more debug in utils

nqb commented on certificate.go:
```diff
@@ -90,33 +92,27 @@ func addCertToMachine(userCertDecode string, CERTUTIL_PROGRAM_PATH string) error
-											os.Remove(userCertDecode)
-											os.Remove("profile.xml")
-											log.Fatal("Invalid certificate: ", exitStatus)
+											addNewLinesToDebug(T("invalidCertificate"))
```
There is no `exitStatus` string here.

JeGoi replied:
Yes, You are right. Debug is the Object where debug lines are printed.
When debug is enabled, the program exit is a "Close" button.
I removed the log.Fatal to be able to view logs in debug mode and change the way the program exit.
The viewErrorAndExit is when Debug is disable. It shows an error Messagebox then Exit. Maybe, I should rename that function.

JeGoi replied:
addNewLinesToDebug() and viewErrorAndExit() are closed but addNewLinesToDebug have more info. I could merge them to have just one function when both are called one after the other.

julsemaan replied:
One call would be better indeed

JeGoi replied:
Done

julsemaan commented on certificate.go:
```diff
@@ -90,33 +92,27 @@ func addCertToMachine(userCertDecode string, CERTUTIL_PROGRAM_PATH string) error
 											mw.Close()
 										case int(ERROR_INVALID_DATA):
-											walk.MsgBox(windowMsgBox, T("errorWindowTitle"), T("invalidCertificate"), walk.MsgBoxOK)
-											os.Remove(userCertDecode)
```
I agree that this massive copy/paste needs to be removed but I don't see it in the defer so that it still happens when this is done

This would mean the files are left there after it exits unless I'm missing where this has been added

JeGoi replied:
All files are now removed on exit.
In utils.go, func cleanTmpFiles

nqb reviewed (commented):
I find `addNewLinesToDebug` function name not really appropriate.
If I'm not wrong, purpose of this function is:
- in debug, to print text in a specific manner (certainly through a specific window)
or
- to print text

JeGoi committed:
Fix warnings, Merge exit in simple function

JeGoi committed:
Add code to exit
Fix command line
Fix mix template vs profile
Remove extra file creation
Workflow more accurate

JeGoi committed:
Fix load Image

JeGoi committed:
Add exit if no image available

JeGoi committed:
Change Close button size

JeGoi committed:
Fix hide close button

nqb commented on utils.go:
```diff
@@ -72,41 +213,32 @@ func decodeCertificate(certificate string) ([]byte, error) {
+	return writeInFIle(myFile, "", decodedCertificate)
+}
+
+func writeInFIle(myFile *os.File, st string, b []byte) error {
```
Should be `writeInFile` no ? Calls to this function need to be ajusted.

JeGoi replied:
Done

nqb commented on parse.go:
```diff
@@ -61,114 +59,180 @@ type Template struct {
+func prepareMainWindow() {
 	currentWorkingDirectory, err := os.Executable()
 	if err != nil {
 		walk.MsgBox(windowMsgBox, "Error", "Unable to get current working directory, please contact your local support.", walk.MsgBoxOK)
```
Do we keep it or it need to be replaced ?

JeGoi replied:
Well, this message box will stop the program before the main window is open. It should be good to keep it in case we do not have access to this directory, and every files created during the process are based on this path.

nqb commented on utils.go:
```diff
@@ -1,65 +1,206 @@
+		addNewLinesToDebug(s + " " + sExtra)
+	}
+	if !debug {
+		walk.MsgBox(windowMsgBox, T("errorWindowTitle"), s+"\r\nPlease enable Debug Mode and contact your local support.", walk.MsgBoxOK)
```
Do we keep it or it needs to be replaced ?

JeGoi replied:
This one is Yes, we need it because it is the message before exit when something fails. Except if you do not want to see something when the program fails.

nqb commented on utils.go:
```diff
@@ -1,65 +1,206 @@
 
+// Chunk large string at x charaters
+// Source: https://stackoverflow.com/a/61469854
+func chunks(s string, chunkSize int) string {
```
Could you rename that function using usual convention for function (verbNoun) ? Perhaps `chunkLargeStrings`

In that function, a variable `chunks` has same name that the function `chunks`.

JeGoi replied:
Done

JeGoi committed:
Fix Functions Name

julsemaan commented:
Please make sure you run all the files through `go fmt` if that's not done already

JeGoi committed:
Fix issue with write/open files

JeGoi committed:
Add test on server response

JeGoi committed:
Fix logs when success
Change Variables to Structures shared across functions
Add catch when return something different than 200

JeGoi committed:
Remove extra Variable
Instanciate WinAgent* Objects

JeGoi committed:
Add message when exit on success

extrafu commented:
This still needs to be merged prior the release.

julsemaan committed:
ERROR_CANCELED is 64 bit long

julsemaan commented:
The agent in this PR is unable to install an EAP-TLS profile since its unable to install the CA

Tested stable agent and this agent with the same configuration, stable works, this doesn't 

Will wait for @JeGoi to look into it, we should shovel this to 11.2 if its not a quick fix

JeGoi committed:
Remove extra error when install ca

JeGoi committed:
Change few debug lines + Add windows version

julsemaan commented:
I was able to test EAP-PEAP successfully but EAP-TLS has failed to install the CA:

```
============== PacketFence Provisioning Agent ==============
User's locale uint16 is: 3084
Language file successfully created.
Program will be translated in french.
The current build windows is after 2004 version: 19044
Welcome to PF debug
PNG file pf_bg.png successfully created at C:\Users\prime\Ap
pData\Local\Temp\pf_bg.png
Been able to attach background image.
Extract Wireless configuration profile: 13
L'Autorité de Certification n'a pas pu être installée sur vo
tre machine, contactez votre support local. exit status 0x80
070003
Erreur lors de l'ouverture du fichier d'Autorité de Certific
ation, contactez votre support local. open : Le fichier spéc
ifié est introuvable.
Unable to get CA fingerprint. open : Le fichier spécifié est
 introuvable.
executetemplateSuccess
EAP Type is: 13
Le fichier de profil a été créé avec succès.
Success adding profile The wireless profile was successfully
 added to the machine.
Please select your newly added profi
le Boulette-Secure in the WiFi networks. 
```
